components/navigation/MobileNavbar.tsx

```diff
@@ -8,9 +7,57 @@ import SearchRouteIcon from '../icons/navigation/SearchRouteIcon';
 import AboutRouteIcon from '../icons/navigation/AboutRouteIcon';
 import AddItemRouteIcon from '../icons/navigation/AddItemRouteIcon';
 import MessageRouteIcon from '../icons/navigation/MessageRouteIcon';
+import NotificationDot from '../NotificationDot';
+
+import { usePathname } from 'next/navigation';
+import { useEffect, useState } from 'react';
+import selectUserUnreadConversations from '@/supabase/models/messaging/selectUserUnreadConversations';
+import selectLoggedUserId from '@/supabase/utils/selectLoggedUserId';
+import notificationWatcher from '@/supabase/channels/notificationsWatcher';
 
 const MobileNavbar = () => {
   const pathname = usePathname();
+  const [userId, setUserId] = useState<string>('');
+  const [hasNotification, setHasNotification] = useState<boolean>(false);
+
+  useEffect(() => {
+    const getUserId = async () => {
+      try {
+        const data = await selectLoggedUserId();
+        if (!data) return;
+        setUserId(data);
+      } catch (error) {
+        console.error('Error fetching user Id:', error);
+      }
+    };
+    getUserId();
+  }, []);
+
+  useEffect(() => {
+    const getUnreadConversations = async () => {
+      if (!userId) return;
+
+      try {
+        const unreadConversations = await selectUserUnreadConversations(userId);
+        if (unreadConversations.length > 0 && pathname !== '/conversations') {
+          setHasNotification(true);
+        }
+      } catch (error) {
+        console.error('Failed to fetch unread conversations :', error);
+      }
+    };
+    getUnreadConversations();
+  }, [userId]);
+
+  useEffect(() => {
+    if (pathname === '/conversations') {
+      setHasNotification(false);
+    }
+  }, [pathname]);
+
+  useEffect(() => {
+    notificationWatcher(userId, pathname, setHasNotification);
+  }, [hasNotification, pathname, userId]);
```

Comment:
we can maybe get rid of that first `useEffect` by sticking the pathname evaluation into the second one ? (I haven't tested that idea 😅 )

Reply:
Yay one less `useEffect` good thinking :+1: